devices/temp_nix.go

```diff
@@ -4,13 +4,50 @@
 package devices
 
 import (
+	"log"
+
+	"github.com/anatol/smart.go"
+	"github.com/jaypipes/ghw"
+	"github.com/jaypipes/ghw/pkg/block"
 	"github.com/shirou/gopsutil/host"
 )
 
+var disks []block.Disk
+var smDevices []smart.Device
+
 func init() {
 	devs() // Populate the sensorMap
+	RegisterStartup(startBlock)
 	RegisterTemp(getTemps)
 	RegisterDeviceList(Temperatures, devs, defs)
+	RegisterShutdown(endBlock)
+}
+
+func startBlock(vars map[string]string) error {
+	block, err := ghw.Block()
+	if err != nil {
+		log.Print("error getting block device info")
+		return err
+	}
+	for _, disk := range block.Disks {
+		dev, err := smart.Open("/dev/" + disk.Name)
+		if err == nil {
+			disks = append(disks, *disk)
+			smDevices = append(smDevices, dev)
+		}
+	}
+	return err
+}
+
+func endBlock() error {
+	for _, dev := range smDevices {
+		err := dev.Close()
+		if err != nil {
+			log.Print("error closing device")
+			return err
```

Comment:
i do not think it makes sense to return here. The method must close all devices (except those that failed).